docs/explainer.md

```diff
@@ -1,137 +1,1058 @@
+
+AXML's `<import>` and `<template is=...` elements.
+
+```xml
+<import src="./header.axml" />
+
+<template is="header" data="{{...item}}"/>
+```
+
+[More information](https://opendocs.alipay.com/mini/framework/axml-template).
+
+##### Baidu Smart Mini Program
+
+Declaration in the configuration (`.json`) document. 
+
+```js
+// home.json
+{
+  "usingComponents": {
+     "custom": "/components/custom/custom"
+   }
+}
+```
+
+And declaration in the SWAN document using the ID in the configuration file.
+
+```xml
+<!-- home.swan -->
+<view>
+   <custom name="swanapp"></custom>
+</view>
+```
+
+[More information](https://smartprogram.baidu.com/docs/develop/framework/dev/).
+
+
+#### Component properties (definition)
+
+How the properties (attributes) of the components are defined in each MiniApp version. 
+
+##### Quick App (Xiaomi, Huawei)
+
+Components define properties in a `props` array within the object declared in the `<script>` section.     
+
+```html
+<!-– comp.ux -->
+<script>
+  export default {
+    props: ['say', 'propObject'],
+    onInit() {
+      console.info(this.say);    
+      console.info(this.propObject)
+    }
+  }
+</script>
+```
+[More information](https://doc.quickapp.cn/tutorial/framework/parent-child-component-communication.html).
+
+##### Alipay Mini Program
+
+Definition of the component using `Component()` and the properties as pairs in the `props` object.
+
+```js
+// /components/index/index.js
+Component ({
+  props : { 
+    name : 'My name'}
+});
+```
+[More information](https://opendocs.alipay.com/mini/framework/component_object).
+
+
+##### Baidu Smart Mini Program
+
+Definition of the component using `Component()` and the properties in the `properties` object.
+
+```js
+// (custom.js)
+Component({
+  properties: {
+    name: {
+      type: String,
+      value: 'swan',
+    }
+  }
+})
+```
+
+[More information](https://smartprogram.baidu.com/docs/develop/framework/custom-component/).
+
+
+#### Component arguments
+
+How the properties (attributes) are passed to the components when new instances are created. 
+
+##### Quick App (Xiaomi, Huawei)
+
+Use the defined `props` directly as instance attributes.
+
+```xml
+<import name="comp" src="./comp"></import>
+
+<template>
+  <div>
+    <comp 
+      say="First Test" 
+      prop-object="{{obj}}">
+    </comp>
+  </div>
+</template>
+```
+
+[More information](https://doc.quickapp.cn/tutorial/framework/parent-child-component-communication.html).	
+
+
+##### Alipay Mini Program
+
+Pass the properties as instance attributes.
+
+```xml
+<!-- /pages/index/index.axml -->
+<comp name='Another name'></comp> 
+```
+
+[More information](https://opendocs.alipay.com/mini/framework/component_object).	
+
+##### Baidu Smart Mini Program
+
+Properties as instance attributes
+
+```js
+// home.json
+{
+  "usingComponents": {
+     "custom": "/components/custom/custom"
+   }
+}
+```
+
+```xml
+// .swan document
+<view>
+    <custom name="Passing text" >
+    </custom>
+</view>
+```
```

Comment:
In my opinion, we may need to add how to use arguments, which makes this use case more complete.


```js
// home.json
{
  "usingComponents": {
     "custom": "/components/custom/custom"
   }
}
```

```xml
// .swan document
<view>
    <custom name="Passing text" >
    </custom>
</view>
```

```xml
// .swan custom component
<view>
    {{name}}     // rendering "Passing text"
</view>
```

Reply:
I agree. This is only the explainer. I've opened several PRs, one for each section. 

We can include the usage of these parameters in the PR #17 . Please, note that in the first draft I only referenced the use of  the `<slot>` for parameters because I think it's common in all the implementations (and similar to the standard  HTML).  We also could refer from #15 (where we define the properties and basic properties of components).

Reply:
Agree with you, too!